src/platform/lumin-runtime/elements/builders/model-builder.js

```diff
@@ -168,6 +175,36 @@ export class ModelBuilder extends RenderBuilder {
         }
     }
 
+    _createSpinner (prism) {
+        const spinner = this._createNode(ui.UiLoadingSpinner, 'Create', prism, ui.LoadingSpinnerType.k2dSpriteAnimation);
+        const [w, h] = spinner.getSize();
+        const [x, y, z] = spinner.getLocalPosition();
+
+        spinner.setLocalPosition([x - (w / 2), y - (h / 2), z]);
+        return spinner;
+      }
+
+      async _downloadResource (url, path, element, prism, importScale) {
+        // Set color mask
+        element.setColor([0.1, 0.1, 0.1, 0.1]);
```

Comment:
What does setColor do for the model (which does not exist yet)?

Reply:
The framework creates a model but with INVALID_RESOURCE_ID. I wanted to avoid having bright shape in front of the user while downloading. However, if we add the option to provide placeholder asset while downloading this code will be removed.